Update Open Drone ID documentation (#477)
mavlink/mavlink-devguide: en/services/opendroneid.md

# Open Drone ID (WIP)

> Direct Remote Identification (DRI) or Remote ID (RID) is/will be a mandatory technology for Unmanned Aircrafts (UA) in Japan, the United States of America and the European Union.

Reviewer: ```suggestion
Direct Remote Identification (DRI) or Remote ID (RID) is/will be a mandatory technology for Unmanned Aircraft (UA) in Japan, the United States of America and the European Union.
```

Author: Done

> > **Note** The Open Drone ID messages are tagged in the definition file as "work in progress".

Reviewer: No longer true.

> > **Note** The Open Drone ID messages are tagged in the definition file as "work in progress".
>   They may still change and should not be used in production environments.

Reviewer: Not true any more
```suggestion
```

Therefore the relevant data is divided into different categories and each category is transmitted via its own message.

The standards defines 6 such messages and an additional 7th message type for packing multiple messages together into a message pack (used when transmitting on Wi-Fi NaN, Wi-Fi Beacon or Bluetooth Long Range with Extended Advertising).
To support easy data transfers to/from a Remote ID (RID) transmitter/receiver component, MAVLink messages supporting all the fields of the drone ID messages have been made available.
See [Messages](#messages) below.

## Use case examples {#use_cases}

> There are multiple possible use cases for the MAVLink drone ID messages:

Reviewer: Which of the use cases listed below are mandatory to be supported for the RID compliance? Worth mentioning?

Author: The only real mandatory use case is that the UA broadcasts valid ID etc. data. How that data is distributed inside the UAS doesn't really matter from a regulatory perspective. There is no requirement to use MAVLink messages, so none of the listed use cases are mandatory.

* A flight controller sends ID, location etc. data to an onboard Bluetooth/Wi-Fi RID transmitter component.

> * An onboard Bluetooth/Wi-Fi receiver picks up drone ID messages from surrounding aircrafts, relays this information using MAVLink drone ID messages to the flight controller, which then uses the information e.g. for Detect And Avoid (DAA) calculations.

Reviewer: ```suggestion
* An onboard Bluetooth/Wi-Fi receiver picks up drone ID messages from surrounding aircraft, relays this information using MAVLink drone ID messages to the flight controller, which then uses the information e.g. for Detect And Avoid (DAA) calculations.
```

  Slight compression is applied.
  Example code for this conversion can be found in the project: [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c).

The [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c) contains code for decoding the MAVLink messages and "compressing" the data into data structures for transmission over Bluetooth or Wi-Fi (or vice-versa).

The standards do not impose any requirements for a drone to be capable of receiving drone ID messages, nor any requirements for reacting to their content (requirements from local legislation might be different).

An example Android receiver implementation for broadcast drone ID messages is [available here](https://play.google.com/store/apps/details?id=org.opendroneid.android_osm) ([source codes](https://github.com/opendroneid/receiver-android)).

Code related to (Internet) Network Remote ID can be found in the [InterUSS Project](https://github.com/interuss) and https://github.com/uastech/standards (Unofficial reference for UAS-related APIs).

## Message update rates {#update_rates}

> The standards require that the Location message is broadcast/published from the transmitter at least once per second.

Author: You are right. The ambiguity of how to refer clearly to certain entities has been bothering me the entire time when writing all of the documentation on that page. Every component in the UAS is transmitting and receiving, one way or the other.

I will try to do some improvements.

The rest of the messages must be broadcast/published once per 3 seconds (the rules in some regions have tightened this requirement to 1 second for also the Basic ID and the System message.
Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).

Not all message types or fields within a message type are mandatory to broadcast/publish.
The mandatory message set vary from region to region.
Please see the [summary](https://github.com/opendroneid/opendroneid-core-c#comparison).

The standards require that the data contained in any transmitted/published message (on the air/internet) is not older than 1 second (e.g. location data, timestamps etc.).

The integrator of the UAS must consider the combination of the above requirements when deciding the internal update rates for each component in the MAVLink network that generate data for the drone ID transmitter.

> It should be expected that there will be some delays from e.g. the GCS sending a System message via the flight controller to the transmitter component, which gather this and data from other messages together and generate the final message data set, which is transmitted/published on the air.

Reviewer 2: I think it's pretty clear the GCS needs to send OPEN_DRONE_ID_SYSTEM faster than 1 Hz in order to meet the "not older than 1 second requirement", is this saying the integrator is required to have their own custom build of Mission Planner/QGC etc?

Author: It is not totally clear to me, on how abstract a level to keep this description. I don't really want to start going into details of specific control stacks (ArduPilot, PX4). I guess both teams will now work on a full-scale integration of drone ID and set suitable values for the various components that are being tested. However, anyone is free to replace any of these components with other variants and that person then ends up being the system integrator. 

I will try to think of a way to clarify that a bit.

Reviewer: From a mavlink perspective there is an assumption that streaming rates are guidelines, and that often these aren't really thought about in detail because we assume that people will be doing some level of manual tuning in order to not swamp their channels. 

However if there is a hard requirement for certification then there should be a note to this effect on particular messages - not necessarily of the "number" but of the existence of the number.

You should avoid details of the implementation if at all possible. What should be the same is the "API" - so all information in the messages should be the same, and if messages cause something to happen, that behaviour should be the same.

Hope that makes sense.

Authentication | Not specified (reserved) | [OPEN_DRONE_ID_AUTHENTICATION](../messages/common.md#OPEN_DRONE_ID_AUTHENTICATION) | Provides authentication data for the UA.
Self-ID | Self-ID | [OPEN_DRONE_ID_SELF_ID](../messages/common.md#OPEN_DRONE_ID_SELF_ID) | Plain text message that can be used by Operators to identify themselves and the purpose of an operation. Can also be used to provide optional additional clarification in an emergency/remote ID system failure situation.
<a id="OPEN_DRONE_ID_SYSTEM"></a>System | System | [OPEN_DRONE_ID_SYSTEM](../messages/common.md#OPEN_DRONE_ID_SYSTEM) | Includes Remote Pilot location, multiple aircraft information (group), if applicable, and additional system information.
Operator ID | Operator ID | [OPEN_DRONE_ID_OPERATOR_ID](../messages/common.md#OPEN_DRONE_ID_OPERATOR_ID) | Provides the Operator ID.
Message Pack | Message Pack | [OPEN_DRONE_ID_MESSAGE_PACK](../messages/common.md#OPEN_DRONE_ID_MESSAGE_PACK) | A payload mechanism for combining the messages above into a single message pack. Used with Bluetooth Extended Advertising, Wi-Fi NaN and Wi-Fi Beacon.

> - | - | [OPEN_DRONE_ID_ARM_STATUS](../messages/common.md#OPEN_DRONE_ID_ARM_STATUS) | Broadcast once per second from the RID transmitter/receiver components, to indicate their state. Similar to [Heartbeat](#heartbeat), but with additional text information in the error field when they are not ready to arm.

Reviewer: Might be better like this:

```suggestion
- | - | [OPEN_DRONE_ID_ARM_STATUS](../messages/common.md#OPEN_DRONE_ID_ARM_STATUS) | Sent by RID transmitter/receiver to indicate that the RID system is "ready to use". This should be used as an arming condition for the flight stack. Note that this differs from the [HEARTBEAT](#heartbeat) which indicates that the component is "alive" but not necessarily ready to use.
```

> - | - | [OPEN_DRONE_ID_SYSTEM_UPDATE](../messages/common.md#OPEN_DRONE_ID_SYSTEM_UPDATE) | A subset of the [OPEN_DRONE_ID_SYSTEM](#OPEN_DRONE_ID_SYSTEM) message, containing only fields that must be updated regularly by the GCS. Can optionally be used instead of the full `OPEN_DRONE_ID_SYSTEM` message, to reduce the traffic on the control link.

Reviewer: I'm not certain the semantics are quite "right" here. What this says is that you can send OPEN_DRONE_ID_SYSTEM_UPDATE  _instead_ of OPEN_DRONE_ID_SYSTEM. 

That is possibly true if none of the information in OPEN_DRONE_ID_SYSTEM that is not in OPEN_DRONE_ID_SYSTEM_UPDATE  is mandatory or desired. 

My assumption is that you want the information in both, but that OPEN_DRONE_ID_SYSTEM_UPDATE  contains the stuff that updates at high rate and for which there is possibly also some compliance requriement? Not sure because you refer to **GCS at the target below**.

Anyway, this should probably be reworded to say that if you use only OPEN_DRONE_ID_SYSTEM it must be sent at high rate for compliance. If you use both messages you can send the more efficient OPEN_DRONE_ID_SYSTEM_UPDATE  at high rate and stream OPEN_DRONE_ID_SYSTEM at a much lower rate.

Also clarify if GCS is the destination.

Author: Good points. I have tried to clarify the text.
The GCS is the originator.

> **Note** The raw byte layout of the MAVLink messages is not exactly the same as what a RID Bluetooth/Wi-Fi transmitter component will broadcast over the air.
  Slight compression is applied.
  Example code for this conversion can be found in the project: [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c).

The [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c) contains code for decoding the MAVLink messages and "compressing" the data into data structures for broadcast over Bluetooth or Wi-Fi (or vice-versa for reception).

> The standards do not impose any requirements for a drone to be capable of receiving drone ID messages, nor any requirements for reacting to their content (requirements from local legislation might be different).

Reviewer: Just checking this is true - the standards are only related to broadcast of position?

Author: That is correct. It is essentially a license plate. E.g. cars must display them but are not required to go around and read the license plates of other cars.

You are allowed to implement receivers in your UAS (DAA would be the most obvious reason) but there are no regulatory rules mandating you to do this.

> An example Android receiver implementation for broadcast drone ID messages is [available here](https://play.google.com/store/apps/details?id=org.opendroneid.android_osm) ([source codes](https://github.com/opendroneid/receiver-android)).

Reviewer: ```suggestion
An example Android receiver implementation for broadcasting drone ID messages is [available here](https://play.google.com/store/apps/details?id=org.opendroneid.android_osm) ([source codes](https://github.com/opendroneid/receiver-android)).
```

Author: This was actually correct, but I admit, misleading if you are not familiar with the Broadcast vs Internet RID terms.

I changed it to "An example Android receiver implementation for Bluetooth/Wi-Fi drone ID is"

Code related to (Internet) Network Remote ID can be found in the [InterUSS Project](https://github.com/interuss) and https://github.com/uastech/standards (Unofficial reference for UAS-related APIs).

> ## Message update rates {#update_rates}

Reviewer: This is all about the standards and what might change. As a reader I'm more interested in clear instructions. 

So 
1. Point out the drivers
2. List the expected rates for known standards based on what PX4 and ArduPilot are actually setting for each case
4. A note that these are all guidelines and that systems will still need to be validated.

This is just my thinking. You could argue that this approach discourages users actually checking the standards, which they must do. But I was hoping we could give setting that will nominally "do the job" and that might not have to be changed.

--- 

MAVLink message streaming rates are largely driven by the requirements for compliance to various [standards](#specifications) (including ASTM F3411 and ASD-STAN prEN 4709-00). Generally rates should be tuned to the lowest rate that guarantees meeting the standards in the region in which the system is to be used.

For compliance with ASTM F3411 and ASD-STAN prEN 4709-00 commonly used rates are:
- OPEN_DRONE_ID_SYSTEM_UPDATE : 2 Hz  [this is  a made-up list]
- OPEN_DRONE_ID_SYSTEM_UPDATE: 0.1 Hz (1 Hz if OPEN_DRONE_ID_SYSTEM_UPDATE is not used)
- ETC
- 
 In addition  the following rates are recommended, but not required for compliance:
- OPEN_DRONE_ID_ARM_STATUS: 1Hz.

Note all messages are mandatory. Implementers are expected to check relevant standards blah blah.

Author: I agree with you but right now I don't know what exact rates will be used in Ardupilot and PX4. We will need to do this update later on, once the implementation work settles down.

> The standards require that the Location message is broadcast/published from the RID transmitter component at least once per second.

Reviewer: "The standards" us quite broad and might change - say Australia adds a standard. We should at least link to which ones we mean here and now. I presume we mean the stuff in specifications?

```suggestion
ASTM F3411 and ASD-STAN prEN 4709-00 (and possibly other [standards ](#specifications)) require that the Location message is broadcast/published from the RID transmitter component at least once per second.
```

Author: Added

The rest of the messages must be broadcast/published once per 3 seconds (the rules in some regions have tightened this requirement to 1 second for also the Basic ID and the System message.
Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).

Not all message types or fields within a message type are mandatory to broadcast/publish.
The mandatory message set vary from region to region.
Please see the [summary](https://github.com/opendroneid/opendroneid-core-c#comparison).

> The standards require that the data contained in any broadcasted/published message (on the air/internet) is not older than 1 second (e.g. the location data, timestamps etc.).

Reviewer: So for messages published every 3 seconds the data must not be more than 1 second old?

Author: Sorry, that is actually wrong. Good that you pointed out this. It only applies to the fields in the Location message. I will correct this.

I now realise that the standards are actually missing a requirement for how old the operator location/altitude and the full timestamp in the System message must be. I think we all just implicitly assumed that the 1 second freshness rule applied to fields in all messages.

I had already written the below when I found out the above. I will keep it here just for educational purposes.

Originally, the messages that only require 3 seconds between them, all contained fairly static data. The UA ID, the operator ID etc. never changes. The exception is the System message, which contains the operator location. I assume that it was decided way back (even before I joined the ASTM standard working group) that it is enough to update the operator location every third second.

However, the argument might still go that internally in the UAS, the GCS must send the operator location once per second to the RID transmitter component so that the operator location that is transmitted on the air is never older than one second (even though that only happens every third second).

A very last minute change in the ASTM F3211-22 update was done earlier this year, to meet the FAA requirements. A full timestamp field had to be added somewhere and the only place it would fit was in the System message. However, for the US, the MoC document (ASTM F3586-22) had already overriden the 3 second requirement and mandated 1 second refresh cycles for the Basic ID, the Location and the System message. I.e. for the US, this is quite clear (1 second for everything) and for Europe, the System timestamp is not a mandatory field, so sending System every third second doesn't break any requirement.

Reviewer 2: The MOC seems pretty clear, but I think most explicitly in the testing section, specifically:

> 8.6.2.4 Pass Criteria—For each message type, the time interval between two messages is calculated, and the interval is compared against the maximum as specified in Specification F3411, Subsection 5.4.4.1, and as amended by MOC 5.1.3.8. The interval shall be less than or equal to the allowed time (1 second). In summary, all fields in Table 2 shall be received at least every second.

Reviewer 2: Also 

> 8.9.5.2 Pass Criteria—Each increment shall receive the required messages at least every second for 9 seconds out of 10 seconds and no more than four increments may fail this ratio, and no two failures may be adjacent to one another.

Author: The problem, which I only just now discovered, is that you can be compliant with all of the above by just sending the same non-changing data at a high rate. 

Clearly the intention is to get updated operator location and timestamp data but none of the above require the data to be fresh. Only that the transmitter component must ping out whatever (old) data it happens to have once per second.

This is clearly an oversight that no-one discovered during the standardization work.

Reviewer 2: Isn't 6.1.3 pretty clear?

> Additionally, the requirement in MOC 5.1.3.8, following Part 89, effectively mandates that all three required messages are transmitted within 1 second, with this method resulting in **the system timestamp being associated with all required data elements is never older than 1 second.**

Author: Those two places might "save" us here but they aren't exactly crystal clear requirements.

@gabrielcox: Do you have any opinions on this? Clearly the intention in the back of the head of all of us was that both GCS location and the system timestamp should be refreshed once per second but we somehow failed to document it clearly. See the discussion above.

Author: Gabriel and I had a chat about this and agreed that the the various places that Tim points out above are enough to cover things. The FAA rule is very clear on this topic so there isn't really any incentive for anyone to try to nitpick through the exact wording in the MoC and claim that they can be compliant without having the fields in the System message less than a second old.

[§89.310 h (4)](https://www.regulations.gov/document/FAA-2019-1100-53264): The unmanned aircraft must broadcast the latitude, longitude, and geometric altitude of the unmanned aircraft and its control station no later than 1.0 seconds from the time of measurement to the time of broadcast.

I will update the documentation accordingly and mark this as resolved.

To be compliant in different regions and receive certification acceptance, the UAS as a complete entity must obey the above timing restrictions.
Although popular autopilot SW stacks will be verified to be capable of this, the compliance testing will have been done using a certain set of HW/SW components.
Any deviation from this set, will require the integrator of the UAS to consider the combination of the above requirements when deciding the internal update rates for each component in the MAVLink network, generating data for the RID transmitter component.

It must be expected that there will be some delays from e.g. the GCS sending a System message via the flight controller to the RID transmitter component.

> The RID transmitter component will gather this data plus data from other messages together and generate the final message data set, which is broadcasted on the air.

Reviewer: ```suggestion
The RID transmitter component will gather this data plus data from other messages together and generate the final message data set, which is broadcast over the air.
```

Author: Done

Each of these components will be running on their own update and forward/relay cycle and can thus introduce delays.
I.e. running everything at exactly one second cycles, probably will lead to violations of the data "freshness" requirement.
But going to the other extreme of forcing all messages to be at 10+ Hz would likely lead to unnecessary airwave noise on the (Command and Control) C2 link/Wi-Fi/Bluetooth spectrum.

> Some suitable balance of refresh rates for each component involved must be found.

Reviewer: ```suggestion
Components are expected to tune refresh rates to ensure compliance without saturating internal communication links.
```

Author: Changed

Authentication | Not specified (reserved) | [OPEN_DRONE_ID_AUTHENTICATION](../messages/common.md#OPEN_DRONE_ID_AUTHENTICATION) | Provides authentication data for the UA.
Self-ID | Self-ID | [OPEN_DRONE_ID_SELF_ID](../messages/common.md#OPEN_DRONE_ID_SELF_ID) | Plain text message that can be used by Operators to identify themselves and the purpose of an operation. Can also be used to provide optional additional clarification in an emergency/remote ID system failure situation.
<a id="OPEN_DRONE_ID_SYSTEM"></a>System | System | [OPEN_DRONE_ID_SYSTEM](../messages/common.md#OPEN_DRONE_ID_SYSTEM) | Includes Remote Pilot location, multiple aircraft information (group), if applicable, and additional system information.
Operator ID | Operator ID | [OPEN_DRONE_ID_OPERATOR_ID](../messages/common.md#OPEN_DRONE_ID_OPERATOR_ID) | Provides the Operator ID.
Message Pack | Message Pack | [OPEN_DRONE_ID_MESSAGE_PACK](../messages/common.md#OPEN_DRONE_ID_MESSAGE_PACK) | A payload mechanism for combining the messages above into a single message pack. Used with Bluetooth Extended Advertising, Wi-Fi NaN and Wi-Fi Beacon.

> - | - | [OPEN_DRONE_ID_ARM_STATUS](../messages/common.md#OPEN_DRONE_ID_ARM_STATUS) | Broadcasted once per second from the RID transmitter/receiver components, to indicate their state. Similar to [Heartbeat](#heartbeat), but with additional text information in the error field when they are not ready to arm.
> - | - | [OPEN_DRONE_ID_SYSTEM_UPDATE](../messages/common.md#OPEN_DRONE_ID_SYSTEM_UPDATE) | A subset of the SYSTEM message, containing only fields that must be updated regularly by the GCS. Can optionally be used instead of the full SYSTEM message, to reduce the traffic on the control link.

Reviewer: Better linking
```suggestion
- | - | [OPEN_DRONE_ID_ARM_STATUS](../messages/common.md#OPEN_DRONE_ID_ARM_STATUS) | Broadcast once per second from the RID transmitter/receiver components, to indicate their state. Similar to [Heartbeat](#heartbeat), but with additional text information in the error field when they are not ready to arm.
- | - | [OPEN_DRONE_ID_SYSTEM_UPDATE](../messages/common.md#OPEN_DRONE_ID_SYSTEM_UPDATE) | A subset of the [OPEN_DRONE_ID_SYSTEM](#OPEN_DRONE_ID_SYSTEM) message, containing only fields that must be updated regularly by the GCS. Can optionally be used instead of the full `OPEN_DRONE_ID_SYSTEM` message, to reduce the traffic on the control link.
```

> System | System | [OPEN_DRONE_ID_SYSTEM](../messages/common.md#OPEN_DRONE_ID_SYSTEM) | Includes Remote Pilot location, multiple aircraft information (group), if applicable, and additional system information.

Reviewer: Add anchor for each cross linking

```suggestion
<a id="OPEN_DRONE_ID_SYSTEM"></a>System | System | [OPEN_DRONE_ID_SYSTEM](../messages/common.md#OPEN_DRONE_ID_SYSTEM) | Includes Remote Pilot location, multiple aircraft information (group), if applicable, and additional system information.
```

Component/System | Description
--- | ---
[MAV_COMP_ID_AUTOPILOT1](../messages/common.md#MAV_COMP_ID_AUTOPILOT1) | The flight controller/autopilot. Knows the ID of the UA, the current location, altitude, speed etc.
Ground Control Station | GCS with a human user interface for inputting the operator ID, text description of the flight purpose, method for obtaining the operator location etc.
[MAV_COMP_ID_ODID_TXRX_1](../messages/common.md#MAV_COMP_ID_ODID_TXRX_1) | A Remote ID transmitter/receiver component (Bluetooth/Wi-Fi/Internet).
[MAV_COMP_ID_ODID_TXRX_2](../messages/common.md#MAV_COMP_ID_ODID_TXRX_2) | A Remote ID transmitter/receiver component (Bluetooth/Wi-Fi/Internet).
[MAV_COMP_ID_ODID_TXRX_3](../messages/common.md#MAV_COMP_ID_ODID_TXRX_3) | A Remote ID transmitter/receiver component (Bluetooth/Wi-Fi/Internet).

The autopilot/flight controller is typically the component that knows about the data needed for the Basic ID and the Location messages.

> It must regularly send MAVLink messages with this information to the RID transmitter component.

Reviewer: Just a suggestion.

```suggestion
It must stream MAVLink messages with this information to the RID transmitter component.
```

Author: Done

The method for the transmitters to identify MAVLink messages from the GCS, is described in the [Heartbeat](#heartbeat) section below.

The components in the UAS involved in the drone ID MAVLink message exchange, must keep track of the Heartbeat MAVLink messages from the other components.
If the Heartbeat messages are not received within a required time interval, they must declare a malfunction of the Remote ID system and indicate this in the `status` field of the [Location](https://mavlink.io/en/messages/common.html#OPEN_DRONE_ID_LOCATION) message.
This must also be indicated to the operator of the UAS.

Similarly, the autopilot and GCS components must listen to the [OPEN_DRONE_ID_ARM_STATUS](../messages/common.md#OPEN_DRONE_ID_ARM_STATUS) from the RID transmitter component(s) and not allow the UA to be airborne before the RID transmitter component(s) is ready.

> During flight, if the arm status indicates a failure, similar action must be taken as for a lack of Heartbeat messages.

Reviewer: ```suggestion
During flight, if the arm status indicates a failure, similar action must be taken as for a lack of HEARTBEAT messages.
```

Author: Done

> There is no need for the GCS to listen to drone ID MAVLink messages.

Author: Correct. I forgot about this case. I will update.

I guess the GCS actually needs to listen to both the heartbeat and the arm status. Similarly for the auto-pilot entity.

### Heartbeat {#heartbeat}

Each component involved in the MAVLink message exchange, is required to regularly send out MAVLink [HEARTBEAT](../messages/common.md#HEARTBEAT) messages in order to facilitate discovery and monitoring of the component in the UAS.

> Please see further details in the [Heartbeat documentation](heartbeat.md).

Reviewer: ```suggestion
Please see further details in the [HEARTBEAT](heartbeat.md) documentation.
```

Author: Done

The MAVLink [HEARTBEAT](../messages/common.md#HEARTBEAT) message serves as the way for RID transmitter/receiver components to identify the `sysid` of the GCS.
The GCS will send out MAVLink [HEARTBEAT](../messages/common.md#HEARTBEAT) messages with its `sysid` [field](https://github.com/ArduPilot/pymavlink/blob/master/generator/C/include_v2.0/mavlink_types.h#L115) set to the GCS's system ID and the `type` set to [MAV_TYPE_GCS](../messages/common.md#MAV_TYPE_GCS).
The RID transmitter/receiver components must interpret all MAVLink Open Drone ID messages from that system ID, as coming from the GCS.
There is no dedicated component ID for GCSs, hence the system ID must be used instead for identifying the GCS.

## Possible future improvements {#improvements}

The current set of MAVLink messages do not provide any means for controlling some of the RID transmitter/receiver component details.
The RID transmitter/receiver component must be hard-coded by the manufacturer to a certain configuration.
Some discussion and proposal for this type of messages can be found [here](https://github.com/mavlink/mavlink/pull/1865).
It would be useful to control e.g. the following items:

> * Starting/stopping braodcast

Reviewer: ```suggestion
* Starting/stopping broadcast
```

Author: Done

* Configure the broadcast method (BT4, BT5, Beacon, NaN)
* Wi-Fi channel configuration for Beacon
* Message update rates on the air

> No region currently require drone ID publication via the internet (Network Remote ID).

Reviewer: ```suggestion
No regions currently require drone ID publication via the internet (Network Remote ID).
```

Author: Done
